Exporting an item from workspace or workflow to Simple Archive Format throws errors in DSpace 8 and 9. The items are nevertheless correctly exported.

## Describe the bug
Exporting an item from workspace or workflow to Simple Archive Format throws errors in DSpace 8 and 9. The items are nevertheless correctly exported. The problem is that `ItemExportServiceImpl.java` tries to write the optional collections file, while an owning collection is not yet set for workspace and workflow items.

This are the error messages:

```
java.lang.Exception: java.lang.NullPointerException: Cannot invoke "org.dspace.content.Collection.getHandle()" because the return value of "org.dspace.content.Item.getOwningCollection()" is null
	at org.dspace.app.itemexport.ItemExport.internalRun(ItemExport.java:178)
	at org.dspace.scripts.DSpaceRunnable.run(DSpaceRunnable.java:150)
	at org.dspace.app.launcher.ScriptLauncher.executeScript(ScriptLauncher.java:154)
	at org.dspace.app.launcher.ScriptLauncher.handleScript(ScriptLauncher.java:132)
	at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:99)
Caused by: java.lang.NullPointerException: Cannot invoke "org.dspace.content.Collection.getHandle()" because the return value of "org.dspace.content.Item.getOwningCollection()" is null
	at org.dspace.app.itemexport.ItemExportServiceImpl.writeCollections(ItemExportServiceImpl.java:367)
	at org.dspace.app.itemexport.ItemExportServiceImpl.exportItem(ItemExportServiceImpl.java:178)
	at org.dspace.app.itemexport.ItemExportServiceImpl.exportItem(ItemExportServiceImpl.java:151)
	at org.dspace.app.itemexport.ItemExportCLI.process(ItemExportCLI.java:64)
	at org.dspace.app.itemexport.ItemExport.internalRun(ItemExport.java:174)
	... 4 more
```


## To Reproduce
Steps to reproduce the behavior:
1. Add a new item and keep it in workspace or workflow
2. Export this item via the command line tool like this: 
```
./dspace export -t ITEM -i <ID or handle of item to export> -d <destination where you want items to go> -n 1
```
3. Observe the error messages as well es the correctly exported item



## Expected behavior
Exporting of items in workspace or workflow should not print error messages.
As the `collections` information is marked as optional, it should be enough to just not write collection information, if it is not available.

```
collections  -- (Optional) text file that contains the handles of the collections the item will belong to. Each handle in a row.
             -- Collection in first line will be the owning collection.
```
https://wiki.lyrasis.org/display/DSDOC8x/Importing+and+Exporting+Items+via+Simple+Archive+Format#ImportingandExportingItemsviaSimpleArchiveFormat-DSpaceSimpleArchiveFormat

## Solution
My colleague @marsaoua already has a solution. One of us will open a pull request shortly.

--- a/dspace-api/src/main/java/org/dspace/app/itemexport/ItemExportServiceImpl.java
+++ b/dspace-api/src/main/java/org/dspace/app/itemexport/ItemExportServiceImpl.java
@@ -352,7 +352,7 @@ public class ItemExportServiceImpl implements ItemExportService {
 
     /**
      * Create the 'collections' file.  List handles of all Collections which
-     * contain this Item.  The "owning" Collection is listed first.
+     * contain this Item. The "owning" Collection is listed first.
      *
      * @param item list collections holding this Item.
      * @param destDir write the file here.
@@ -363,12 +363,14 @@ public class ItemExportServiceImpl implements ItemExportService {
         File outFile = new File(destDir, "collections");
         if (outFile.createNewFile()) {
             try (PrintWriter out = new PrintWriter(new FileWriter(outFile))) {
-                String ownerHandle = item.getOwningCollection().getHandle();
-                out.println(ownerHandle);
+                Collection owningCollection = item.getOwningCollection();
+                // The owning collection is null for workspace and workflow items
+                if (owningCollection != null) {
+                    out.println(owningCollection.getHandle());
+                }
                 for (Collection collection : item.getCollections()) {
-                    String collectionHandle = collection.getHandle();
-                    if (!collectionHandle.equals(ownerHandle)) {
-                        out.println(collectionHandle);
+                    if (!collection.equals(owningCollection)) {
+                        out.println(collection.getHandle());
                     }
                 }
             }

